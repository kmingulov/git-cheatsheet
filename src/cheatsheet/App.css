.header {
  padding-left: 15px !important;
  padding-right: 15px !important;
}

.mainContent {
  padding-top: 20px;
}

.git-command-column {
  width: 40%;
}

/* .git { } */

/* .git-command { } */

/* .git-remote { } */

.git-url {
  text-decoration: underline;
}

.git-branch {
  font-weight: bold;
}

/* .git-commit { } */

/* .git-ref { } */

.git-file {
  font-style: italic;
}
