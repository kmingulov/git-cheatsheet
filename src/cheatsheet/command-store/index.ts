export * from './CommandStore';
export * from './lunr';
