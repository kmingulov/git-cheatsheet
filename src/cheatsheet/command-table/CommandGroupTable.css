h4.command-group-title {
  pointer-events: none;
}

h4.command-group-title:before {
  display: block;
  height: 90px;
  margin-top: -90px;
  visibility: hidden;
  content: '';
}

h4.command-group-title > div {
  pointer-events: auto;
}

h4.command-group-title:hover .command-group-link {
  visibility: visible;
}

.command-group-link {
  visibility: hidden;
}

.command-group-link:hover {
  text-decoration: none;
}
