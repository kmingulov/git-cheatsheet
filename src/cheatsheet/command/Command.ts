import { CommandPart } from './CommandPart';

export class Command {
  public readonly scmBreezeShortcut: Array<CommandPart | string>;
  public readonly command: Array<CommandPart | string>;
  public readonly description: Array<CommandPart | string>;

  constructor(scmBreezeShortcut: Array<CommandPart | string>,
              commandParts: Array<CommandPart | string>,
              descriptionParts: Array<CommandPart | string>) {
    this.scmBreezeShortcut = scmBreezeShortcut;
    this.command = commandParts;
    this.description = descriptionParts;
  }
}
