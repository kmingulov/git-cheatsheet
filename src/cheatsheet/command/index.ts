export * from './Command';
export * from './CommandGroup';
export * from './CommandPart';
export * from './CommandPartType';
