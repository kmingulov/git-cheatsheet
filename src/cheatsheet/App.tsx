import React, { Component, ReactElement } from 'react';
import { Container, FormControl, InputGroup, Navbar } from 'react-bootstrap';
import { FaSearch } from 'react-icons/fa';

import { CommandStore } from './command-store';
import { CommandTable } from './command-table';

import commands from './command-data/commands';

import './App.css';

const store = new CommandStore(commands);

const initialState = { searchTerm: '' };
type State = Readonly<typeof initialState>;

class App extends Component<object, State> {
  public readonly state: State = initialState;

  public render(): ReactElement {
    return (
      <div>
        <Navbar bg='primary' sticky='top'>
          <Container className='header'>
            <Navbar.Brand>Git Cheatsheet</Navbar.Brand>
          </Container>
        </Navbar>

        <Container className='mainContent'>
          <InputGroup className='searchBox'>
            <InputGroup.Prepend>
              <InputGroup.Text>
                <FaSearch/>
              </InputGroup.Text>
            </InputGroup.Prepend>
              <FormControl
                placeholder='Searchâ€¦'
                onChange={ (event: any) => this.setState({ searchTerm: event.target.value }) }
              />
          </InputGroup>

          <CommandTable store={ store } searchTerm={ this.state.searchTerm }/>
        </Container>
      </div>
    );
  }
}

export default App;
