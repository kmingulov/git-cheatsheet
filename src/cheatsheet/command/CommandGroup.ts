import { Command } from './Command';

/**
 * Class representing a semantically bound group of commands.
 */
export class CommandGroup {
  /** Group title. */
  public readonly title: string;
  /** Array of commands. */
  public readonly commands: ReadonlyArray<Command>;

  /**
   * Constructs a new {@link CommandGroup}.
   * @param title group title
   * @param commands array of commands
   */
  constructor(title: string, commands: ReadonlyArray<Command>) {
    this.title = title;
    this.commands = commands;
  }
}
