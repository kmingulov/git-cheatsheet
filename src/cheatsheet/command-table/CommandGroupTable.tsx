import React, { Component, ReactElement } from 'react';
import { Badge, Table } from 'react-bootstrap';

import { CommandGroup } from 'cheatsheet/command/CommandGroup';
import { CommandTableRow } from './CommandTableRow';

interface CommandGroupTableProps {
  group: CommandGroup;
}

export class CommandGroupTable extends Component<CommandGroupTableProps> {
  public render(): ReactElement {
    const { group } = this.props;

    return (
      <div>
        <h4>
          { group.title }
          { ' ' }
          <Badge variant='secondary'>{ group.commands.length }</Badge>
        </h4>
        <Table striped bordered>
          <tbody>
            { group.commands.map((command, i) => (
              <CommandTableRow command={ command } key={ i }/>
            ))}
          </tbody>
        </Table>
      </div>
    );
  }
}
