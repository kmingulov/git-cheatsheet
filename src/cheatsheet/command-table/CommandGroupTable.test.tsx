import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';

import { Command, CommandGroup } from 'cheatsheet/command';

import { CommandGroupTable } from './CommandGroupTable';

Enzyme.configure({adapter: new Adapter()});

jest.mock('./CommandTableRow');

const cmd = new Command('cmd-1', [], [], []);
const otherCmd = new Command('cmd-2', [], [], []);
const group = new CommandGroup('Basic Commands', [cmd, otherCmd]);

describe(CommandGroupTable, () => {
  it('renders correctly', () => {
    const component = mount(<CommandGroupTable group={ group }/>);
    expect(component).toMatchSnapshot();
  });
});
