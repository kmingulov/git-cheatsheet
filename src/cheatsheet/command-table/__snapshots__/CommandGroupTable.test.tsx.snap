// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommandGroupTable re-renders when props are changed 1`] = `
<CommandGroupTable
  group={
    CommandGroup {
      "commands": Array [
        Command {
          "command": Array [],
          "description": Array [],
          "id": "cmd-1",
          "scmBreezeShortcut": Array [],
        },
        Command {
          "command": Array [],
          "description": Array [],
          "id": "cmd-2",
          "scmBreezeShortcut": Array [],
        },
      ],
      "id": "basic",
      "title": "Basic Commands",
    }
  }
>
  <div
    className="command-group"
  >
    <h4
      className="command-group-title"
      id="basic"
    >
      <div>
        Basic Commands
         
        <ForwardRef(Bootstrap(Badge))
          variant="secondary"
        >
          <Badge
            bsPrefix="badge"
            pill={false}
            variant="secondary"
          >
            <span
              className="badge badge-secondary"
            >
              2
            </span>
          </Badge>
        </ForwardRef(Bootstrap(Badge))>
         
        <a
          className="command-group-link"
          href="#basic"
        >
          #
        </a>
      </div>
    </h4>
    <ForwardRef(Bootstrap(Table))
      bordered={true}
      className="command-group-table"
      striped={true}
    >
      <Table
        bordered={true}
        bsPrefix="table"
        className="command-group-table"
        striped={true}
      >
        <table
          className="table command-group-table table-striped table-bordered"
        >
          <tbody>
            <CommandTableRow
              command={
                Command {
                  "command": Array [],
                  "description": Array [],
                  "id": "cmd-1",
                  "scmBreezeShortcut": Array [],
                }
              }
              key="0"
            />
            <CommandTableRow
              command={
                Command {
                  "command": Array [],
                  "description": Array [],
                  "id": "cmd-2",
                  "scmBreezeShortcut": Array [],
                }
              }
              key="1"
            />
          </tbody>
        </table>
      </Table>
    </ForwardRef(Bootstrap(Table))>
  </div>
</CommandGroupTable>
`;

exports[`CommandGroupTable re-renders when props are changed 2`] = `
<CommandGroupTable
  group={
    CommandGroup {
      "commands": Array [],
      "id": "another",
      "title": "Another Group",
    }
  }
>
  <div
    className="command-group"
  >
    <h4
      className="command-group-title"
      id="another"
    >
      <div>
        Another Group
         
        <ForwardRef(Bootstrap(Badge))
          variant="secondary"
        >
          <Badge
            bsPrefix="badge"
            pill={false}
            variant="secondary"
          >
            <span
              className="badge badge-secondary"
            >
              0
            </span>
          </Badge>
        </ForwardRef(Bootstrap(Badge))>
         
        <a
          className="command-group-link"
          href="#another"
        >
          #
        </a>
      </div>
    </h4>
    <ForwardRef(Bootstrap(Table))
      bordered={true}
      className="command-group-table"
      striped={true}
    >
      <Table
        bordered={true}
        bsPrefix="table"
        className="command-group-table"
        striped={true}
      >
        <table
          className="table command-group-table table-striped table-bordered"
        >
          <tbody />
        </table>
      </Table>
    </ForwardRef(Bootstrap(Table))>
  </div>
</CommandGroupTable>
`;

exports[`CommandGroupTable renders correctly 1`] = `
<CommandGroupTable
  group={
    CommandGroup {
      "commands": Array [
        Command {
          "command": Array [],
          "description": Array [],
          "id": "cmd-1",
          "scmBreezeShortcut": Array [],
        },
        Command {
          "command": Array [],
          "description": Array [],
          "id": "cmd-2",
          "scmBreezeShortcut": Array [],
        },
      ],
      "id": "basic",
      "title": "Basic Commands",
    }
  }
>
  <div
    className="command-group"
  >
    <h4
      className="command-group-title"
      id="basic"
    >
      <div>
        Basic Commands
         
        <ForwardRef(Bootstrap(Badge))
          variant="secondary"
        >
          <Badge
            bsPrefix="badge"
            pill={false}
            variant="secondary"
          >
            <span
              className="badge badge-secondary"
            >
              2
            </span>
          </Badge>
        </ForwardRef(Bootstrap(Badge))>
         
        <a
          className="command-group-link"
          href="#basic"
        >
          #
        </a>
      </div>
    </h4>
    <ForwardRef(Bootstrap(Table))
      bordered={true}
      className="command-group-table"
      striped={true}
    >
      <Table
        bordered={true}
        bsPrefix="table"
        className="command-group-table"
        striped={true}
      >
        <table
          className="table command-group-table table-striped table-bordered"
        >
          <tbody>
            <CommandTableRow
              command={
                Command {
                  "command": Array [],
                  "description": Array [],
                  "id": "cmd-1",
                  "scmBreezeShortcut": Array [],
                }
              }
              key="0"
            />
            <CommandTableRow
              command={
                Command {
                  "command": Array [],
                  "description": Array [],
                  "id": "cmd-2",
                  "scmBreezeShortcut": Array [],
                }
              }
              key="1"
            />
          </tbody>
        </table>
      </Table>
    </ForwardRef(Bootstrap(Table))>
  </div>
</CommandGroupTable>
`;
